utils/oominfo/oomparser.go

```diff
@@ -0,0 +1,200 @@
+package main
+
+import (
+  "bufio"
+  "fmt"
+  "log"
+  "os"
+  "strings"
+  "os/exec"
+  "time"
+  "regexp"
+  "io"
+  "errors"
+)
+
+// struct to hold a list of oomInstance objects
+type oomLogList struct {
+	oomInstances []*oomInstance
+}
+
+// struct that contains information related to an OOM kill instance
+type oomInstance struct {
+	pid string
+	processName string
+	timeOfDeath time.Time
+	containerName containerLevel
+}
+
+// recursive struct used to hold container and subcontainer names
+type containerLevel struct {
+	name string
+	subcontainer *containerLevel
+}
+
+/* tries to populate oomInstance object from a line and add it to the oomLog
+struct. called by readMessages */
+// http://stackoverflow.com/questions/25382073/defining-golang-struct-function-using-pointer-or-not
+func (oomLog *oomLogList) addToOOMLog(line string) error {
+	linetime, err := getDateFromLine(line)
+	if err != nil {
+		return err
+	}
+	words := strings.Fields(line)
+	for i := 0; i < len(words) - 1; i++ {
+		if words[i] == "Killed" && words[i + 1] == "process"  && i + 3 < len(words) - 1 {
```

Comment:
Consider using regexp for this. You should be able to identify the line and parse the value at once